### PR TITLE
feat: Function assets

### PR DESCRIPTION
**Pull request by gabrielmpinto**
## Why?

Adding function assets to Fleek Functions.
Depend on external library for esbuild configs: https://github.com/fleek-platform/functions-esbuild-config
- No longer polyfills asynclocalstorage
- More Node.js support

## How?

- Depend on https://www.npmjs.com/package/@fleek-platform/functions-esbuild-config
- Optionally upload assets dir and pass the CID to the SDK

## Contribution checklist?

- [x] The commit messages are detailed
- [x] The `build` command runs locally
- [x] Assets or static content are linked and stored in the project
- [x] You have manually tested
- [ ] You have provided tests

## Security checklist?

- [x] Sensitive data has been identified and is being protected properly
- [x] Injection has been prevented (parameterized queries, no eval or system calls)

## Preview?

Optionally, provide the preview url here

### CODE REVIEW

File: src/commands/functions/deploy.ts [hunk 1]
```diff
@@ -145,6 +172,7 @@ const deployAction: SdkGuardedFunction<DeployActionArgs> = async ({
       cid: uploadResult.pin.cid,
       sgx: isSGX,
       blake3Hash,
+      // assetsCid,
```
Comment (kkudryaev) [inline src/commands/functions/deploy.ts]: Should we leave it commented out?

File: locales/en.json [hunk 2]
```diff
@@ -351,5 +351,11 @@
   "warnProvideValidDomainName": "Please provide a valid domain name.",
   "cmdAuthLoginDescription": "Authenticate the CLI session using the Fleek Platform Web app. Open the URL in your favourite browser to initiate the browser-based login process. Select your preferred authentication method and return to CLI once completed.",
   "cmdAuthLogoutDescription": "Ends your active CLI session, securing your account. Disables access to personal features such as storage and site deployment until re-authentication.",
-  "invalidHostname": "Invalid hostname `{hostname}`. Please try again!"
+  "invalidHostname": "Invalid hostname `{hostname}`. Please try again!",
+  "functionsUseAssets": "Path to assets to be uploaded to Fleek Storage along with your function code. Accessible via the global variable `fleek.env.ASSETS_URL`.",
```
Comment (punkbit) [inline locales/en.json]:
```suggestion
  "functionsUseAssets": "Choose the assets directory to upload alongside your function. These assets will be accessible in your code through 'fleek.env.ASSETS_URL', e.g. /User/MyProject/assets."
",
```

File: locales/en.json [hunk 3]
```diff
@@ -351,5 +351,11 @@
...
+  "assetsPathIsNotAFolder": "The provided assets path is not a valid folder",
```
Comment (punkbit) [inline locales/en.json]:
```suggestion
  "assetsPathIsNotAFolder": "The specified path doesn't exist or is not a valid directory. Please provide a valid directory path containing your assets."
",
```

File: locales/en.json [hunk 4]
```diff
@@ -351,5 +351,11 @@
...
+  "uploadingAssets": "Uploading assets...",
+  "assetsUploadSuccess": "Assets uploaded successfully",
+  "uploadAssetsFailed": "Failed to upload assets for function",
```
Comment (punkbit) [inline locales/en.json]:
```suggestion
  "uploadAssetsFailed": "Failed to upload function assets to Fleek Network. Check your network connection and try again.",
```

File: src/commands/functions/deploy.ts [hunk 5]
```diff
@@ -47,12 +51,35 @@ const deployAction: SdkGuardedFunction<DeployActionArgs> = async ({
...
   }
 
+  if (assetsPath && isSGX) {
+    output.error(t('assetsNotSupportedInSgx'));
+    return;
+  }
+
+  let assetsCid: string | undefined = undefined;
+  if (assetsPath) {
+    console.log('assetsPath', assetsPath);
```
Comment (punkbit) [inline src/commands/functions/deploy.ts]:
```suggestion
```

File: src/commands/functions/deploy.ts [hunk 6]
```diff
@@ -47,12 +51,35 @@ const deployAction: SdkGuardedFunction<DeployActionArgs> = async ({
...
+    } else {
+      output.spinner(t('uploadingAssets'));
+      assetsCid = await uploadFunctionAssets({
+        sdk,
+        assetsPath,
+        functionName: functionToDeploy.name,
+      });
+      output.success(t('assetsUploadSuccess'));
+    }
+  }
```
Comment (punkbit) [inline src/commands/functions/deploy.ts]:
```suggestion
if (!await isValidFolder(assetsPath)) {
  output.error(t('assetsPathIsNotAFolder'));
  return;
}

output.spinner(t('uploadingAssets'));
assetsCid = await uploadFunctionAssets({
  sdk,
  assetsPath,
  functionName: functionToDeploy.name,
});
output.success(t('assetsUploadSuccess'));
```

File: src/commands/functions/deploy.ts (hunk 1 above)
Comment (gabrielmpinto) [reply src/commands/functions/deploy.ts]: Nope! Uncommented it, thanks

Comment (punkbit) [inline src/commands/functions/deploy.ts]:
```suggestion
```

File: src/commands/functions/deploy.ts (hunk 6 above)
Comment (gabrielmpinto) [reply src/commands/functions/deploy.ts]: Moved folder validation to the helper method aswell

File: src/commands/functions/utils/uploadFunctionAssets.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+}: {
+  sdk: FleekSdk;
+  functionName: string;
+  assetsPath?: string;
+}): Promise<string | undefined> => {
+  if (!assetsPath) {
+    return;
+  }
+
+  if (!(await isValidFolder(assetsPath))) {
```
Comment (punkbit) [inline src/commands/functions/utils/uploadFunctionAssets.ts:19]:
```suggestion
  if (!await isValidFolder(assetsPath)) {
```

**Review by punkbit — APPROVED** [review]

File: src/commands/functions/utils/uploadFunctionAssets.ts (hunk 7 above)
Comment (gabrielmpinto) [reply src/commands/functions/utils/uploadFunctionAssets.ts:19]: Formatter is adding it

**Pull request merged**